### PR TITLE
add configuration options which allow for deterministic classnames

### PR DESCRIPTION
**Pull request by thehandsomepanther**

### CODE REVIEW

File: test/vudu.js [hunk 1]
```diff
@@ -64,6 +64,22 @@ test('returns an object of values with type string', t => {
...
+    container: {
+      color: 'blue',
+      textAlign: 'left'
+    }
+  };
+
+  v.options({ suffix })(style1, t.context.sheet);
+  t.is(t.context.sheet.cssRules.length, 1);
+  t.true(RegExp(`${suffix}$`).test(t.context.sheet.cssRules[0].selectorText))
+});
```
Comment (hhff) [inline test/vudu.js]: Can we do a test for `suffix can be a function` too?

**Comment by hhff** [comment]
@dhunninghake - can you take a look?

The impetus here is that we are using Google Optimize with an app, and because the "suffix" of each class is random, selectors in the Optimize Editor aren't matched on page refresh.

This allows the developer to override the default behavior of using `lodash.uniqueId` to do basically whatever they want.

@thehandsomepanther also added a currying style invocation to avoid having to do 

```
v({ styles }, null, options)
```
everywhere

File: src/vudu.js [hunk 2]
```diff
@@ -17,38 +17,41 @@ let vuduSheet = createSheet('vSheet');
...
   });
   rules.forEach(r => addRule(formatRule(r.styles), r.vuduClass, sheet, true));
-  return rules.reduce((a, b) => {
-    a[b.classname] = b.vuduClass;
-    return a;
-  }, {});
+  return rules.reduce((a, b) => ({
+    ...a,
+    [b.classname]: b.vuduClass,
+  }), {});
```
Comment (dhunninghake) [inline src/vudu.js]: 👌 👌 👌

File: src/vudu.js [hunk 3]
```diff
@@ -17,38 +17,41 @@ let vuduSheet = createSheet('vSheet');
...
  * @returns {Object}
  */
-const buildRuleset = (group, sheet) => {
+const buildRuleset = (group, sheet, options={}) => {
+  const suffix = typeof options.suffix === 'function' ? options.suffix() : options.suffix;
   const rules = Object.keys(group).map(classname => {
     return {
       classname,
-      vuduClass: `${classname}-${uniqueId()}`,
+      vuduClass: `${classname}-${suffix ? suffix : uniqueId()}`,
```
Comment (dhunninghake) [inline src/vudu.js]: could shorten this: `${suffix || uniqueId()}`

File: src/vudu.js [hunk 4]
```diff
@@ -17,38 +17,41 @@ let vuduSheet = createSheet('vSheet');
  *
  * @param {Object} group
  * @param {Object} sheet
+ * @param {Object} [options={}]
```
Comment (dhunninghake) [inline src/vudu.js]: would love to replace these jsdocs eventually with something more useful. maybe TS or Flow

File: test/vudu.js [hunk 5]
```diff
@@ -64,6 +64,39 @@ test('returns an object of values with type string', t => {
...
+    container: {
+      color: 'yellow',
+      textAlign: 'center'
+    }
+  };
+
+  v.options({ suffix })(style1, t.context.sheet);
+  t.is(t.context.sheet.cssRules.length, 1);
+  t.true(RegExp(`${suffixText}$`).test(t.context.sheet.cssRules[0].selectorText))
+});
```
Comment (dhunninghake) [inline test/vudu.js]: Nice tests!

**Pull request merged**